Library/FormatTests.swift

```diff
@@ -168,59 +168,59 @@ final class FormatTests: TestCase {
       withEnvironment(countryCode: "US") {
-        XCTAssertEqual(Format.currency(1_000, country: .us), "1.000 $")
-        XCTAssertEqual(Format.currency(1_000, country: .ca), "1.000 CA$")
-        XCTAssertEqual(Format.currency(1_000, country: .gb), "1.000 £")
-        XCTAssertEqual(Format.currency(1_000, country: .dk), "1.000 DKK")
-        XCTAssertEqual(Format.currency(1_000, country: .de), "1.000 €")
-        XCTAssertEqual(Format.currency(1_000, country: .jp), "1.000 ¥")
+        XCTAssertEqual(Format.currency(1_000, country: .us), "1000 $")
+        XCTAssertEqual(Format.currency(1_000, country: .ca), "1000 CA$")
+        XCTAssertEqual(Format.currency(1_000, country: .gb), "1000 £")
+        XCTAssertEqual(Format.currency(1_000, country: .dk), "1000 DKK")
+        XCTAssertEqual(Format.currency(1_000, country: .de), "1000 €")
+        XCTAssertEqual(Format.currency(1_000, country: .jp), "1000 ¥")
       }
 
       withEnvironment(countryCode: "CA") {
-        XCTAssertEqual(Format.currency(1_000, country: .us), "1.000 US$")
-        XCTAssertEqual(Format.currency(1_000, country: .ca), "1.000 CA$")
-        XCTAssertEqual(Format.currency(1_000, country: .gb), "1.000 £")
-        XCTAssertEqual(Format.currency(1_000, country: .dk), "1.000 DKK")
-        XCTAssertEqual(Format.currency(1_000, country: .de), "1.000 €")
-        XCTAssertEqual(Format.currency(1_000, country: .jp), "1.000 ¥")
+        XCTAssertEqual(Format.currency(1_000, country: .us), "1000 US$")
+        XCTAssertEqual(Format.currency(1_000, country: .ca), "1000 CA$")
+        XCTAssertEqual(Format.currency(1_000, country: .gb), "1000 £")
+        XCTAssertEqual(Format.currency(1_000, country: .dk), "1000 DKK")
+        XCTAssertEqual(Format.currency(1_000, country: .de), "1000 €")
+        XCTAssertEqual(Format.currency(1_000, country: .jp), "1000 ¥")
       }
 
       withEnvironment(countryCode: "GB") {
-        XCTAssertEqual(Format.currency(1_000, country: .us), "1.000 US$")
-        XCTAssertEqual(Format.currency(1_000, country: .ca), "1.000 CA$")
-        XCTAssertEqual(Format.currency(1_000, country: .gb), "1.000 £")
-        XCTAssertEqual(Format.currency(1_000, country: .dk), "1.000 DKK")
-        XCTAssertEqual(Format.currency(1_000, country: .de), "1.000 €")
-        XCTAssertEqual(Format.currency(1_000, country: .jp), "1.000 ¥")
+        XCTAssertEqual(Format.currency(1_000, country: .us), "1000 US$")
+        XCTAssertEqual(Format.currency(1_000, country: .ca), "1000 CA$")
+        XCTAssertEqual(Format.currency(1_000, country: .gb), "1000 £")
+        XCTAssertEqual(Format.currency(1_000, country: .dk), "1000 DKK")
+        XCTAssertEqual(Format.currency(1_000, country: .de), "1000 €")
+        XCTAssertEqual(Format.currency(1_000, country: .jp), "1000 ¥")
       }
     }
 
     withEnvironment(locale: Locale(identifier: "fr")) {
       withEnvironment(countryCode: "US") {
-        XCTAssertEqual(Format.currency(1_000, country: .us), "1 000 $")
```

Comment:
According to https://en.wikipedia.org/wiki/Non-breaking_space referring to the narrow non-breaking space:
> its width is usually one third of the normal space; in other contexts, its width is about 70% of the normal space but may resemble that of the thin space (U+2009), at least with some fonts.[8] Also starting from release 34 of Unicode Common Locale Data Repository (CLDR) the NNBSP is used in numbers as thousands group separator for French locale.[9].

This appears to have been introduced in iOS 13.